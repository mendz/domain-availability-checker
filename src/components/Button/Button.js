import React from 'react';
import PropTypes from 'prop-types';

import './Button.module.css';

const Button = props => (
   <button {...props}>{props.children}</button>
);

Button.propTypes = {
   onClick: PropTypes.func,
   disable: PropTypes.bool,
   type: PropTypes.string
}

export default Button;