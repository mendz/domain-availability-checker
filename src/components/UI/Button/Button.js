import React from 'react';
import PropTypes from 'prop-types';

import classes from './Button.module.css';

const Button = ({ bigger, active, className, ...props }) => {
  const classArr = [classes.Button, className];

  if (bigger) {
    classArr.push(classes.Bigger);
  }

  if (active) {
    classArr.push(classes.Active);
  }

  return (
    <button className={classArr.join(' ')} {...props}>
      {props.children}
    </button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func,
  bigger: PropTypes.bool,
  className: PropTypes.string,
};

export default Button;
