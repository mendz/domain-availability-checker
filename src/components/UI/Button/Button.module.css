.Button {
   border: none;
   box-shadow: 0 1px 2px rgba(1, 1, 1, 0.5);
   background: linear-gradient(20deg, #418550 2%, #4ca860 50%, #578f0e 100%);
   color: #ffff00;
   padding: 1rem 2rem;
   border-radius: 25%;
   cursor: pointer;
   font-size: 1.2rem;
   transition: all 0.5s ease;
}

.Button:active {
   transform: translateY(2px);
   box-shadow: inset 0 3px 2px 3px rgba(1, 1, 1, 0.5);
}

.Button:disabled {
   cursor: not-allowed;
   background: linear-gradient(20deg, #2d5c37 2%, #3a8149 50%, #355807 100%);
   color: #c7c700;
}