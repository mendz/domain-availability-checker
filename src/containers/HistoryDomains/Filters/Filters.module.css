.Filters {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-content: center;
  align-items: baseline;
  min-height: 40px;
}

.Filters > *:not(:last-child) {
  margin-right: 1rem;
}

.SearchInput {
  align-self: stretch;
  font-size: 0.8rem;
  width: 25%;
}

@media all and (max-width: 600px) {
  .Filters {
    align-content: space-between;
  }
}
