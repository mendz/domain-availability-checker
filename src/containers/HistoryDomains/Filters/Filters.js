import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import {
  resetFilter,
  setFilterType,
  setSearchValue,
} from '../../../store/actions/historyDomains';

import Button from '../../../components/UI/Button/Button';
import SymbolsCheck from '../../../components/UI/SymbolsCheck/SymbolsCheck';

import classes from './Filters.module.css';

const Filters = ({
  searchValue,
  filterType,
  resetFilter,
  setFilterType,
  setSearchValue,
}) => {
  const handleFilter = event => {
    const { target } = event;
    setFilterType(target.dataset.filterType);
  };

  const handleSearchInput = event => {
    const { target } = event;
    setSearchValue(target.value);
  };

  const handleResetFilter = () => {
    resetFilter();
  };

  return (
    <div className={classes.Filters}>
      <Button name="reset-filter" onClick={handleResetFilter}>
        Rest Filter
      </Button>
      <input
        className={classes.SearchInput}
        type="search"
        name="domains-filter"
        id="domains-filter"
        onChange={handleSearchInput}
        placeholder="Search Domains..."
        value={searchValue}
      />
      <label>Filter by: </label>
      <Button
        onClick={handleFilter}
        name="show-all"
        data-filter-type="all"
        active={filterType === 'all' ? true : false}
      >
        All
      </Button>
      <Button
        onClick={handleFilter}
        name="show-available"
        data-filter-type="available"
        active={filterType === 'available' ? true : false}
      >
        <SymbolsCheck type="available" />
      </Button>
      <Button
        onClick={handleFilter}
        name="show-unavailable"
        data-filter-type="unavailable"
        active={filterType === 'unavailable' ? true : false}
      >
        <SymbolsCheck type="unavailable" />
      </Button>
    </div>
  );
};

const mapStateToProps = state => ({
  filterType: state.historyDomains.filterType,
  searchValue: state.historyDomains.searchValue,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ resetFilter, setFilterType, setSearchValue }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Filters);

export { Filters };
