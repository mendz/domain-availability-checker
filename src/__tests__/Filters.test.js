import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { Filters } from '../containers/HistoryDomains/Filters/Filters';
import Button from '../components/UI/Button/Button';

describe('<Filters />', () => {
  let wrapper;
  // because FIlters is a functional component, the except will use this instead wrapper.props()
  const initialProps = {
    searchValue: '',
    filterType: '',
    resetFilter: jest.fn(() => {}),
    setFilterType: jest.fn(() => {}),
    setSearchValue: jest.fn(() => {}),
  };

  beforeEach(() => {
    wrapper = shallow(<Filters {...initialProps} />);
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('click filter by available - ✔ will call the function this.props.setFilterType with the argument: "available"', () => {
    const event = {
      target: {
        dataset: {
          filterType: 'available',
        },
      },
    };
    wrapper
      .find(Button)
      .findWhere(comp => comp.prop('name') === 'show-available')
      .simulate('click', event);
    expect(initialProps.setFilterType).toHaveBeenCalledWith('available');
  });

  it('click filter by unavailable - ✖ will call the function this.props.setFilterType with the argument: "unavailable"', () => {
    const event = {
      target: {
        dataset: {
          filterType: 'unavailable',
        },
      },
    };
    wrapper
      .find(Button)
      .findWhere(comp => comp.prop('name') === 'show-unavailable')
      .simulate('click', event);
    expect(initialProps.setFilterType).toHaveBeenCalledWith('unavailable');
  });

  it('click filter by "all" will call the function this.props.setFilterType with the argument: "all"', () => {
    const event = {
      target: {
        dataset: {
          filterType: 'all',
        },
      },
    };
    wrapper
      .find(Button)
      .findWhere(comp => comp.prop('name') === 'show-all')
      .simulate('click', event);
    expect(initialProps.setFilterType).toHaveBeenCalledWith('all');
  });

  it('change the search input will call the function this.props.setSearchValue with the correct argument', () => {
    wrapper.find('input').simulate('change', { target: { value: 's' } });
    expect(initialProps.setSearchValue).toHaveBeenCalledWith('s');

    wrapper.find('input').simulate('change', { target: { value: '' } });
    expect(initialProps.setSearchValue).toHaveBeenCalledWith('');
  });

  it('click the reset button will call the function this.props.resetFilter', () => {
    wrapper
      .find(Button)
      .findWhere(comp => comp.prop('name') === 'reset-filter')
      .simulate('click');
    expect(initialProps.resetFilter).toBeCalled();
  });
});
