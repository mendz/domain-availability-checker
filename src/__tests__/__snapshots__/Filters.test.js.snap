// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> matches the snapshot 1`] = `
<div
  className="Filters"
>
  <Button
    name="reset-filter"
    onClick={[Function]}
  >
    Rest Filter
  </Button>
  <input
    className="SearchInput"
    id="domains-filter"
    name="domains-filter"
    onChange={[Function]}
    placeholder="Search Domains..."
    type="search"
    value=""
  />
  <label>
    Filter by: 
  </label>
  <Button
    active={false}
    data-filter-type="all"
    name="show-all"
    onClick={[Function]}
  >
    All
  </Button>
  <Button
    active={false}
    data-filter-type="available"
    name="show-available"
    onClick={[Function]}
  >
    <SymbolsCheck
      type="available"
    />
  </Button>
  <Button
    active={false}
    data-filter-type="unavailable"
    name="show-unavailable"
    onClick={[Function]}
  >
    <SymbolsCheck
      type="unavailable"
    />
  </Button>
</div>
`;
